web/src/main/java/com/microsoft/applicationinsights/web/extensibility/modules/WebSessionTrackingTelemetryModule.java

```diff
@@ -183,98 +145,17 @@ public boolean getIsUserModuleEnabled() {
-            boolean generateNewSessions = Boolean.parseBoolean(argumentsMap.get(GENERATE_NEW_SESSIONS_PARAM_KEY));
-            this.generateNewSessions = generateNewSessions;
-        }
-
-        if (argumentsMap.containsKey(SESSION_TIMEOUT_PARAM_KEY)) {
-            int sessionTimeoutInMinutes = Integer.parseInt(argumentsMap.get(SESSION_TIMEOUT_PARAM_KEY));
-            this.sessionTimeoutInMinutes = sessionTimeoutInMinutes;
-        }
-    }
-
-    private void setSessionCookie(HttpServletRequest req, HttpServletResponse res) {
-        if (res.isCommitted()) {
-            InternalLogger.INSTANCE.error("Response already committed by a different component. Failed to set session cookie.");
-
-            return;
-        }
-
-        RequestTelemetryContext context = ThreadContext.getRequestTelemetryContext();
-
-        int sessionTimeout = getSessionTimeout(req);
-        SessionContext sessionContext = getTelemetrySessionContext(context);
-        Cookie cookie = HttpCookieFactory.generateSessionHttpCookie(context, sessionContext, sessionTimeout);
-
-        res.addCookie(cookie);
-    }
-
-    /**
-     * If the session timeout has been set by the user via AI config file, then this timeout will be returned.
-     * Otherwise, it will look for a session timeout configured in the application descriptor (web.xml).
-     * Otherwise, default timeout will be returned.
-     */
-    private int getSessionTimeout(ServletRequest servletRequest) {
-        if (sessionTimeoutInMinutes != null) {
-            return sessionTimeoutInMinutes;
-        }
-
-        Integer sessionTimeout = ServletUtils.getRequestSessionTimeout(servletRequest);
-        if (sessionTimeout == null) {
-            sessionTimeout = SessionCookie.SESSION_DEFAULT_EXPIRATION_TIMEOUT_IN_MINUTES;
-        }
-
-        synchronized (this) {
-            sessionTimeoutInMinutes = sessionTimeout;
-        }
-
-        return sessionTimeout;
     }
 
     private SessionContext getTelemetrySessionContext(RequestTelemetryContext aiContext) {
         return aiContext.getHttpRequestTelemetry().getContext().getSession();
     }
 
-    private void startNewSession(RequestTelemetryContext aiContext) {
-        String sessionId = LocalStringsUtils.generateRandomId(true);
-
-        SessionContext session = getTelemetrySessionContext(aiContext);
-        session.setId(sessionId);
-
-        aiContext.setSessionCookie(new SessionCookie(sessionId));
-        trackSessionStateWithRequestSessionId(SessionState.Start, sessionId);
-        aiContext.setIsNewSession(true);
-    }
-
-    private void trackSessionStateWithRequestSessionId(SessionState requiredState, String sessionId) {
-        SessionStateTelemetry sessionStateTelemetry = new SessionStateTelemetry(requiredState);
-        sessionStateTelemetry.getContext().getSession().setId(sessionId);
-
-        telemetryClient.track(sessionStateTelemetry);
-    }
-
     private boolean isNewUser(HttpServletRequest request) {
```

Comment:
This is an interesting case: if no user cookie exists than we should ignore the session cookie as well (and open a bug on the JS team). Reminder: we have a backlog item to validate that both IDs are sent together.